chore: update readme with getting started (#13)
mParticle/crossplatform-sdk-tests: README.md

`models`
`testing` -> `api` -> `models`

#### Writing Tests

> Tests are located in the `Tests` directory

Reviewer: ```suggestion
Tests are located in the `Tests` directory.
```

> End-to-End tests often utilize the `MockServer` api to either assert against certain messages reaching the server or simply blocking until a request is sent/received

Reviewer: ```suggestion
End-to-End tests often utilize the `MockServer` API to either assert against certain messages reaching the server or simply blocking the message until a request is sent/received.
```

Reviewer: I'm not sure it's accurate, but I adde "the message" to clarify what was being blocked.

Author: Ah, good suggestion on the clarification. In this case, we are actually blocking "the thread", so I will apply and update

> A new `MockServer` instance is automatically initialized before every single test, as long as your test class extends one of the base classes

Reviewer: ```suggestion
A new `MockServer` instance is automatically initialized before every single test, as long as your test class extends one of the base classes.
```

> `BaseTest`: a fresh `MockServer` instance, mParticle SDK has not already been started when the tests executes

Reviewer: ```suggestion
* `BaseTest`: a fresh `MockServer` instance. mParticle SDK has not already been started when the tests execute.
```

> `BaseStartedTest`: a fresh `MockServer` instance, and a fully initialized (initial `identify()` call has completed`) mParticle SDK instance

Reviewer: ```suggestion
* `BaseStartedTest`: a fresh `MockServer` instance. A fully initialized mParticle SDK instance (initial `identify()` call has completed) .
```

The `MockServer` instance is stored in a singleton, `server`. Tests should only interact with their `MockServer` instance through this singleton, attempting to invoke the class directly **will** lead to threading/freezing erros when run on iOS.

> ##### Assertions

Reviewer: ```suggestion
##### Assertions

```

> Tests may use the `MockServer` to assert against either

Reviewer: ```suggestion
Tests may use the `MockServer` to assert against one of the following:
```

> - "current" requests state (have/have not already received a matching request), or

Reviewer: ```suggestion
* "current" requests state (have/have not already received a matching request)
```

> - "future" requests state (will/will not be received).

Reviewer: ```suggestion
* "future" requests state (will/will not be received)
```

> For the latter case, you can also choose to block until the assertion passes or continue the test logic and check it's state at a later point. In order to avoid timing issues for "future" blocking assertions for "future" assertions, the test may also specify the "trigger" they wish to test in the `after { }` block

Reviewer: ```suggestion
For the future reuqests state, you can also choose to block until the assertion passes or continue the test logic and check its state at a later point. In order to avoid timing issues for "future" blocking assertions for "future" assertions, the test may also specify the "trigger" they wish to test in the `after { }` block
```

Reviewer: I was confused by "for future blocked assertions for future assertions," but if a developer wouldn't be confused, no need to modify. But check and make sure that one of those 'assertions' isn't meant to be 'request'?

Author: Ah, I think was rephrasing that and ended up leaving duplicates in there.. I'm going to update it to just "for future assertions"

**log an event and block until it is received**
```kotlin
Server
    .endpoint(EndpointType.Events)
    .assertWillReceive { messageBatch: Request<BatchMessage> ->
        messageBatch.body.messages
            .filterIsInstance<MPEvent>()
            .any {
                it.eventName == "my event" && it.eventType == EventType.Social
            }
    }
    .after { mParticle.logEvent(MPEvent("my event", EventType.Social)) }
    .blockUntilFinished()
```

> 

Reviewer: ```suggestion
```

> In the previous test, we had to

Reviewer: ```suggestion
In the previous test we performed the following tasks:

```

1) select an EndpointType. Each endpoint type will automatically populate the following methods with the correct request/response type
    - `Events`
    - `Alias`
    - `Identity_{Login|Logout|Identify|Modify}`
    - `Config`

> 2) select an "assertion type"

Reviewer: ```suggestion
2. Select an "assertion type"
```

    - `assert{Will|WillNot}Receiver { }`
    - `assert{Has|HasNot}Received { }`

> 3) *if Will/WillNot*, define a "trigger" in the `after { }` block

Reviewer: ```suggestion
3. *If Will/WillNot*, define a "trigger" in the `after { }` block.
```

Reviewer: I'm not sure what "If Will/WillNot" is. Should the italics start with Will?

> 4) select a continuation behavior:

Reviewer: ```suggestion
4. Select a continuation behavior:
```

>     - `blockUntilFinished()`: will block and timeout after 5 seconds the assertion is not satisfied.

Reviewer: ```suggestion
    - `blockUntilFinished()`: will block and timeout after 5 seconds if the assertion is not satisfied.
```

>     - `assertFinished()`: will check if the assertion has been satisfied and immediately fail if it has not

Reviewer: ```suggestion
    - `assertFinished()`: will check if the assertion has been satisfied and immediately fail if it has not.
```

>     - `cancel()`: will cancel any timeout timer, if one has been set

Reviewer: ```suggestion
    - `cancel()`: will cancel any timeout timer, if one has been set.
```

>     - `latch`: will return an `MPLatch` instance which can be used to block at any arbitrary point in the test. This instance will automatically un-block if the assertion is satisfied

Reviewer: ```suggestion
    - `latch`: will return an `MPLatch` instance which can be used to block at any arbitrary point in the test. This instance will automatically un-block if the assertion is satisfied.
```

> 

Reviewer: ```suggestion
```

> ##### Request/Response Logic

Reviewer: ```suggestion
##### Request/Response Logic

```

Tests may also set custom request/response logic via the `MockServer`:

> **On the next config request, return a custom response

Reviewer: ```suggestion
**On the next config request, return a custom response**
```

```kotlin
Server
   .endpoint(EndpointType.Config)
   .nextResponse {
       SuccessResponse {
           ConfigResponseMessage(workspaceToken = "some token")
       }
   }
```

> **for the next Events request, send a 404 failure response**

Reviewer: ```suggestion
**For the next Events request, send a 404 failure response**
```
